src/components/chart components/chartsFunctions.js

```diff
@@ -0,0 +1,109 @@
+};
+
+const monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+
+const getDatesInInterval = (startDate, endDate) => {
+    let datesArray = [];
+    for (let i = new Date(startDate); i <= new Date(endDate); i.setDate(i.getDate() + 1)) {
+        datesArray.push(new Date(i));
+    }
+    return datesArray;
+};
+
+const generateLabels = (timePeriod, startDate = null, endDate = null) => {
+    let array = [];
+    if (timePeriod === "Last 10 Days") {
+        for (let i = 0; i < 10; i++) {
+            array[i] = daysLabelDateTransformation(new Date(dateFromThePast(i)));
+        }
+    } else if (timePeriod === "Last Month") {
+        for (let i = getNumberOfDaysOfThePreviousMonth(); i > 0; i--) {
+            array[getNumberOfDaysOfThePreviousMonth() - i] = i;
+        }
+    } else if (timePeriod === "Last Year" || timePeriod === "Two Years Ago") {
+        for (let i = monthNames.length - 1; i >= 0; i--) {
+            array[i] = monthNames[monthNames.length - i - 1];
+        };
+    } else if (timePeriod === "Custom Interval") {
+        return (startDate !== null && endDate !== null && getDatesInInterval(startDate, endDate));
+    }
+    return array.reverse();
+};
+
+const checkForDateOccurrence = (date, sampleData) => {
+    let occurrence = 0;
+    for (let i = 0; i < sampleData.length; i++) {
+        if (date === sampleData[i].date) {
+            occurrence++;
+        }
+    }
+    return occurrence;
+};
+
+const generateChartData = (timePeriod, sampleData, startDate = null, endDate = null) => {
+    let array = [];
+    let number = 0;
+
+    if (timePeriod === "Last 10 Days") {
+        number = 10;
+        for (let i = 0; i < number; i++) {
+            array[i] = checkForDateOccurrence(dateFromThePast(i), sampleData);
+        }
+    }
+    else if (timePeriod === "Last Month") {
+        number = getNumberOfDaysOfThePreviousMonth();
+        for (let i = 0; i < number; i++) {
+            array[i] = checkForDateOccurrence(dateFromThePast(i, lastDayOfPreviousMonth), sampleData);
+        }
+    }
+    else if (timePeriod === "Last Year" || timePeriod === "Two Years Ago") {
+        array = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+        let number = array.length;
+        for (let i = 0; i < sampleData.length; i++) {
+            for (let j = 0; j < number; j++) {
+                if (new Date(sampleData[i].date).getMonth() === j && new Date(sampleData[i].date).getFullYear() === (timePeriod === "Last Year" ? lastYear : secondToLastYear)) {
+                    array[j] = array[j] + 1;
+                    break;
+                }
+            }
+        }
+    } else if (timePeriod === "Custom Interval") {
+        let dates = getDatesInInterval(startDate, endDate);
+        array.length = dates.length;
+        for (let i = 0; i < dates.length; i++) {
+            array[i] = 0;
+            for (let j = 0; j < sampleData.length; j++) {
```

Comment:
this is duplicated code, please consider to move it to a single parametrized function